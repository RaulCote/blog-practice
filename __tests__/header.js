import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../src/components/header';
import { render } from '@testing-library/react';
import { renderLightTheme } from '../utils/tests/themeContext';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderLightTheme(
      <Header siteTitle="Default Starter" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const Title = () => <h1 data-testid="hero-title">React Testing Library</h1>;

describe('React Test Library', () => {
  it('works as expected', () => {
    const { getByTestId } = render(<Title />);

    expect(getByTestId('hero-title')).toHaveTextContent(
      'React Testing Library'
    );
  });
});

describe('GlobalStyle, lets see', () => {
  it('renders with GlobalStyle included!', () => {
    const tree = renderLightTheme(
      <Header sitTitle="Default Starter" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
