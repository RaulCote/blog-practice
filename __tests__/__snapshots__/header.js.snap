// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyle, lets see renders with GlobalStyle included! 1`] = `
<header
  style={
    Object {
      "alignItems": "center",
      "background": "rebeccapurple",
      "display": "flex",
      "justifyContent": "spaceBetween",
      "marginBottom": "1.45rem",
    }
  }
>
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "1.45rem 1.0875rem",
      }
    }
  >
    <h1
      style={
        Object {
          "margin": 0,
        }
      }
    >
      <a
        href="/"
        style={
          Object {
            "color": "white",
            "textDecoration": "none",
          }
        }
      >
        
      </a>
    </h1>
  </div>
</header>
`;

exports[`Header renders correctly 1`] = `
<header
  style={
    Object {
      "alignItems": "center",
      "background": "rebeccapurple",
      "display": "flex",
      "justifyContent": "spaceBetween",
      "marginBottom": "1.45rem",
    }
  }
>
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "1.45rem 1.0875rem",
      }
    }
  >
    <h1
      style={
        Object {
          "margin": 0,
        }
      }
    >
      <a
        href="/"
        style={
          Object {
            "color": "white",
            "textDecoration": "none",
          }
        }
      >
        Default Starter
      </a>
    </h1>
  </div>
</header>
`;
